aip/auth/4117.md

```diff
@@ -0,0 +1,387 @@
+
+### Prerequisite
+
+In order to use workload identity federation to access Google cloud resources
+from non-Google cloud platforms, the following steps are needed to configure
+workload identity pools, providers, service account impersonation and generate
+the JSON configuration file to be used by the auth libraries.
+
+- [Configure Workload Identity Federation from AWS][0]
+- [Configure Workload Identity Federation from Microsoft Azure][1]
+- [Configure Workload Identity Federation from an OIDC identity provider][2]
+- [Configure Workload Identity Federation from a SAML identity provider][3]
+
+### Configuration File Generation and Usage
+
+After workload identity federation is configured, the JSON configuration file
+will need to be generated. This sample shows how an AWS configuration file
+is generated:
+
+```bash
+$ gcloud iam workload-identity-pools create-cred-config \
+    projects/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$POOL_ID/providers/$PROVIDER_ID \
+    --service-account=$SERVICE_ACCOUNT_EMAIL \
+    --aws \
+    --output-file=$FILEPATH.json
+```
+
+The following values would need to be replaced:
+
+- **PROJECT_NUMBER**: Project number of the project that contains the workload
+  identity pool.
+- **POOL_ID**: ID of the workload identity pool.
+- **PROVIDER_ID**: ID of the workload identity pool provider.
+- **SERVICE_ACCOUNT_EMAIL**: Email address of the service account to
+  impersonate.
+- **FILEPATH**: File to save configuration to.
+
+The external identities configuration file can be used with
+[Application Default Credentials][6]. In order to use external identities with
+Application Default Credentials, the full path to this file should be stored
+in the `GOOGLE_APPLICATION_CREDENTIALS` environment variable.
+
+```bash
+export GOOGLE_APPLICATION_CREDENTIALS=/path/to/config.json
+```
+
+The library can now automatically choose the right type of client and initialize
+credentials from the context provided in the configuration file:
+
+```python
+import google.auth
+
+credentials, project = google.auth.default()
+```
+
+External account credentials can also be initialized explicitly using the
+generated configuration file.
+
+```python
+# Sample for Azure or OIDC/SAML providers.
+import json
+
+from google.auth import identity_pool
+
+json_config_info = json.loads(function_to_get_json_config())
+credentials = identity_pool.Credentials.from_info(json_config_info)
+scoped_credentials = credentials.with_scopes(
+    ['https://www.googleapis.com/auth/cloud-platform'])
+```
+
+```python
+# Sample for AWS.
+import json
+
+from google.auth import aws
+
+json_config_info = json.loads(function_to_get_json_config())
+credentials = aws.Credentials.from_info(json_config_info)
+scoped_credentials = credentials.with_scopes(
+    ['https://www.googleapis.com/auth/cloud-platform'])
+```
+
+### Expected Behavior
+
+The auth libraries will use the information in the JSON configuration file to
```

Comment:
s/will/\*\*should\*\*/

See 'Requirement keywords' section in https://google.aip.dev/8

Reply:
Done